### PR TITLE
Generate the README's --help output automatically via cargo +nightly dev generate-all

### PR DESCRIPTION
**Pull request by squiddy**
Closes: #1474

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -301,86 +301,126 @@ ruff path/to/code/ --select F401 --select F403
 See `ruff --help` for more:
 
 ```shell
+<!-- Begin auto-generated cli help. -->
```
Comment (charliermarsh) [inline README.md:304]: It looks like these are appearing in the rendered Markdown.

Comment (charliermarsh) [reply README.md:304]: Screen Shot 2022-12-30 at 1 30 31 PM

https://github.com/charliermarsh/ruff/blob/f3ebb168ef6d9ec5f1de59e0ea78a1fbf13c35f0/README.md

Comment (squiddy) [reply README.md:304]: :man_facepalming: Let me fix that.

Comment (charliermarsh) [reply README.md:304]: No prob, I wasn't sure what would happen 😂

File: README.md [hunk 2]
```diff
@@ -300,88 +300,129 @@ ruff path/to/code/ --select F401 --select F403
...
 Usage: ruff [OPTIONS] [FILES]...
 
 Arguments:
   [FILES]...
+          
 
 Options:
       --config <CONFIG>
           Path to the `pyproject.toml` or `ruff.toml` file to use for configuration
+
```
Comment (charliermarsh) [inline README.md:316]: Any idea where all the newlines are coming from?

Comment (squiddy) [reply README.md:316]: Good point, I'll go looking.

File: README.md (hunk 1 above)
Comment (squiddy) [reply README.md:303]: Fixed it.

**Comment by charliermarsh** [comment]
LGTM!

File: README.md (hunk 2 above)
Comment (squiddy) [reply README.md]: Just got back from my deep-dive into clap.

In the end, I just had to go from `render_long_help` to `render_help`. :sweat_smile:

There is some trailing whitespace after the `FILES` lines, but I don't think that's an issue.

Comment (charliermarsh) [reply README.md]: All good. Thanks again :)

Comment (squiddy) [reply README.md]: @charliermarsh I've noticed that the trailing whitespace seems to cause issues. In https://github.com/charliermarsh/ruff/commit/74903f23d69172474116ec3e05d182cdf9179415 it got removed, which fails CI again. Shall we adjust the cmd to strip trailing whitespace from the generated output?

Comment (charliermarsh) [reply README.md]: Hmm yeah, I guess so. I wonder how it got removed? Is it machine-specific? Or did my editor do it automatically, or something?

Comment (charliermarsh) [reply README.md]: Are you working on this? I could do it later if needed.

Comment (charliermarsh) [reply README.md]: I got it :)

**Pull request merged**